packages/commonwealth/client/scripts/views/components/countdown.tsx

```diff
@@ -1,66 +1,36 @@
 import React from 'react';
-
-import type { ResultNode } from 'mithrilInterop';
-import { ClassComponent, redraw } from 'mithrilInterop';
 import moment from 'moment';
-import { blocknumToTime, formatDuration } from 'helpers';
+import { useCountdown } from 'usehooks-ts';
+
+import { blocknumToDuration, formatDuration } from 'helpers';
 
-type CountdownAttrs = {
-  duration?: moment.Duration;
-  includeSeconds?: boolean;
-  time?: moment.Moment;
+type CountdownProps = {
+  duration: number;
 };
 
-export class Countdown extends ClassComponent<CountdownAttrs> {
-  private timer;
-  private timerHandle;
+export const Countdown = (props: CountdownProps) => {
+  const { duration } = props;
 
-  view(vnode: ResultNode<CountdownAttrs>) {
-    const { time, duration, includeSeconds } = vnode.attrs;
-    if (!time && !duration) return;
+  const [count, { startCountdown }] = useCountdown({
+    countStart: duration,
+    intervalMs: 1000,
+  });
 
-    const durationForDisplay = time
-      ? moment.duration(moment(time).diff(moment()))
-      : duration;
+  React.useEffect(() => {
+    startCountdown();
+  }, [startCountdown]);
 
-    return (
-      <span
-        oncreate={() => {
-          this.timerHandle = setInterval(() => {
-            this.timer++;
-            redraw();
-          }, 1000);
-        }}
-        onremove={() => {
-          if (this.timerHandle) {
-            clearInterval(this.timerHandle);
-          }
-        }}
-      >
-        {includeSeconds
-          ? formatDuration(durationForDisplay)
-          : formatDuration(durationForDisplay, false)}
-      </span>
-    );
-  }
-}
+  return (
+    <span>{formatDuration(moment.duration(moment(count).diff(moment())))}</span>
+  );
+};
 
-type CountdownUntilBlockAttrs = {
+type CountdownUntilBlockProps = {
   block: number;
-  includeSeconds?: boolean;
 };
 
-export class CountdownUntilBlock extends ClassComponent<CountdownUntilBlockAttrs> {
-  view(vnode: ResultNode<CountdownUntilBlockAttrs>) {
-    let { includeSeconds } = vnode.attrs;
-    if (!vnode.attrs.block) return;
-    if (includeSeconds === undefined) includeSeconds = true;
+export const CountdownUntilBlock = (props: CountdownUntilBlockProps) => {
+  const { block } = props;
 
-    return (
-      <Countdown
-        time={blocknumToTime(vnode.attrs.block)}
-        includeSeconds={includeSeconds}
-      />
-    );
-  }
-}
+  return <Countdown duration={blocknumToDuration(block)} />;
```

Comment:
Time seems to be inverted here?

<img width="854" alt="Screen Shot 2023-02-15 at 5 20 14 PM" src="https://user-images.githubusercontent.com/16248081/219197319-a890fa56-7f56-4f38-9ad7-949e9afe3748.png">
<img width="441" alt="Screen Shot 2023-02-15 at 5 20 45 PM" src="https://user-images.githubusercontent.com/16248081/219197326-2e780c7c-45bc-4d02-be9c-249ce6a9821a.png">